src/main/java/io/cryostat/net/reports/ArchivedRecordingReportCache.java

```diff
@@ -41,111 +41,84 @@
             return f;
         }
+
         try {
             generationLock.lock();
             // check again in case the previous lock holder already created the cached file
             if (fs.isReadable(dest) && fs.isRegularFile(dest)) {
                 f.complete(dest);
                 return f;
             }
+            logger.trace("Archived report cache miss for {}", recordingName);
 
-            fs.listDirectoryChildren(savedRecordingsPath).stream()
-                    .filter(name -> name.equals(recordingName))
-                    .map(savedRecordingsPath::resolve)
-                    .findFirst()
-                    .ifPresentOrElse(
-                            recording -> {
-                                logger.trace("Archived report cache miss for {}", recordingName);
-                                try {
-                                    Path saveFile =
-                                            subprocessReportGeneratorProvider
-                                                    .get()
-                                                    .exec(
-                                                            recording,
-                                                            dest,
-                                                            Duration.ofMillis(
-                                                                    TimeoutHandler.TIMEOUT_MS))
-                                                    .get();
-                                    f.complete(saveFile);
-                                } catch (Exception e) {
-                                    logger.error(e);
-                                    f.completeExceptionally(e);
-                                    try {
-                                        fs.deleteIfExists(dest);
-                                    } catch (IOException ioe) {
-                                        logger.warn(ioe);
-                                    }
-                                }
-                            },
-                            () ->
-                                    f.completeExceptionally(
-                                            new RecordingNotFoundException(
-                                                    "archives", recordingName)));
-        } catch (IOException ioe) {
-            logger.warn(ioe);
-            f.completeExceptionally(ioe);
+            Path archivedRecording = recordingArchiveHelper.getRecordingPath(recordingName).get();
+            Path saveFile =
+                    subprocessReportGeneratorProvider
+                            .get()
+                            .exec(
+                                    archivedRecording,
+                                    dest,
+                                    Duration.ofMillis(TimeoutHandler.TIMEOUT_MS))
+                            .get();
+            f.complete(saveFile);
+        } catch (ExecutionException | InterruptedException e) {
```

Comment:
I think this can be removed/merged with the `catch (Exception e)` below. Or at least, the `fs.deleteIfExists(dest)` should be performed here as well.